tools/distpartitioning/data_shuffle.py

```diff
@@ -436,14 +439,58 @@ def exchange_feature(
         assert np.all(global_eids == data[constants.GLOBAL_EID][idx1])
         partid_slice = id_lookup.get_partition_ids(global_dst_nids)
 
+    # determine the shape of the feature-data
+    # this is needed to so that ranks where feature-data is not present
+    # should use the correct shape for sending the padded vector.
+    # exchange length here.
+    feat_dim_len = 0
+    if featdata_key != None:
+        feat_dim_len = len(featdata_key.shape)
+    all_lens = allgather_sizes(
+        [feat_dim_len], world_size, num_parts, return_sizes=True
+    )
+    if all_lens[0] <= 0:
```

Comment:
any testcase to cover this scenario?

Reply:
Since we support no. of chunks < no. of partitions, we will now run into situations when reading dataset from the disk some processes (when handling more than one partition per process) will not have any data in partitions other than '0'. In these cases, we use a sentinal value of node and edge_feature_tids as [(0,0)] to indicate that there is no data in these cases. In such cases we will hit this code, as pointed above. 

The existing test cases, where no. of partitions more than one per process will cover this corner case.